packages/@aws-cdk/aws-stepfunctions-tasks/lib/run-batch-job.ts

```diff
@@ -0,0 +1,341 @@
+   * or add new environment variables to it with an environment override.
+   *
+   * @default - No container overrides
+   */
+  readonly containerOverrides?: ContainerOverrides;
+
+  /**
+   * A list of dependencies for the job.
+   * A job can depend upon a maximum of 20 jobs. You can specify a SEQUENTIAL
+   * type dependency without specifying a job ID for array jobs so that each
+   * child array job completes sequentially, starting at index 0.
+   * You can also specify an N_TO_N type dependency with a job ID for array jobs.
+   * In that case, each index child of this job must wait for the corresponding
+   * index child of each dependency to complete before it can begin.
+   *
+   * @default - No dependencies
+   */
+  readonly dependsOn?: JobDependency[];
+
+  /**
+   * The payload to be passed as parametrs to the batch job
+   *
+   * @default - No parameters are passed
+   */
+  readonly payload?: { [key: string]: any };
+
+  /**
+   * The retry strategy to use for failed jobs from this SubmitJob operation.
+   * When a retry strategy is specified here,
+   * it overrides the retry strategy defined in the job definition.
+   *
+   * @default - No retry strategy
+   */
+  readonly retryStrategy?: RetryStrategy;
+
+  /**
+   * The timeout configuration for this SubmitJob operation.
+   *
+   * @default - No timeout
+   */
+  readonly timeout?: Duration;
+
+  /**
+   * The service integration pattern indicates different ways to call TerminateCluster.
+   *
+   * The valid value is either FIRE_AND_FORGET or SYNC.
+   *
+   * @default SYNC
+   */
+  readonly integrationPattern?: sfn.ServiceIntegrationPattern;
+}
+
+/**
+ * A Step Functions Task to run AWS Batch
+ *
+ * @experimental
+ */
+export class RunBatchJob implements sfn.IStepFunctionsTask {
+  private readonly integrationPattern: sfn.ServiceIntegrationPattern;
+
+  constructor(private readonly props: RunBatchJobProps) {
+    this.integrationPattern =
+      props.integrationPattern || sfn.ServiceIntegrationPattern.SYNC;
+
+    const supportedPatterns = [
+      sfn.ServiceIntegrationPattern.FIRE_AND_FORGET,
+      sfn.ServiceIntegrationPattern.SYNC
+    ];
+
+    if (!supportedPatterns.includes(this.integrationPattern)) {
+      throw new Error(
+        `Invalid Service Integration Pattern: ${this.integrationPattern} is not supported to call RunBatchJob.`
+      );
+    }
+  }
+
+  public bind(_task: sfn.Task): sfn.StepFunctionsTaskConfig {
+    return {
+      resourceArn: getResourceArn(
+        'batch',
+        'submitJob',
+        this.integrationPattern
+      ),
+      policyStatements: [
+        new iam.PolicyStatement({
+          resources: ['*'],
+          actions: [
+            'batch:SubmitJob',
+            'batch:DescribeJobs',
+            'batch:TerminateJob'
+          ]
+        }),
+        new iam.PolicyStatement({
+          resources: [
+            `arn:aws:events:${Aws.REGION}:${Aws.ACCOUNT_ID}:rule/StepFunctionsGetEventsForBatchJobsRule`
```

Comment:
Can you explain what this is, what it does and why there's a hardcoded event rule here?

Reply:
The AWS Batch permissions [doc](https://docs.aws.amazon.com/step-functions/latest/dg/batch-iam.html) requires it.

Reply:
That page simply seems to be an example of what a policy could be.

Reply:
As far as I remember, I tried to deploy without this particular statement and got some error around it. I also got the same suggestion for the AWS support in the regard. I will try the deployment again for confirmation without it once.

Reply:
Just confirmed that the deployment fails without the above statement with the below error:
> 'arn:aws:iam::<..>:role/aws-stepfunctions-integ-StateMachineRoleB840431D-1TSHHD7R6HU9A' is not authorized to create managed-rule. (Service: AWSStepFunctions; Status Code: 400; Error Code: AccessDeniedException; Request ID: c6eaea50-cd22-4316-9825-c770c356bad5)